test/toc.spec.ts

```diff
@@ -0,0 +1,194 @@
+
+      await page.evaluate(changeTab, 'muzak')
+
+      await expect(page).toMatchElement('h1', { timeout: 5000 }) // Wait for tab to load
+      await expect(page).not.toMatchElement('h2[id^=_], h3[id^=_]', { timeout: 5000 }) // Beef
+
+      await page.evaluate(changeTab, 'programming')
+
+      await expect(page).toMatchElement('h1', { timeout: 5000 }) // Wait for tab to load
+      await expect(page).not.toMatchElement('h2[id^=_], h3[id^=_]', { timeout: 5000 }) // Beef
+    })
+  })
+
+  describe('Missing heading IDs on Swedish tabs', () => {
+    it('should have zero default ids on swedish tabs', async () => {
+      await page.goto('http://localhost:8080/build/index.html?sv')
+
+      await page.evaluate(changeTab, 'chemistry')
+
+      await expect(page).toMatchElement('h1', { timeout: 5000 }) // Wait for tab to load
+      await expect(page).not.toMatchElement('h2[id^=_], h3[id^=_]', { timeout: 5000 }) // Beef
+
+      await page.evaluate(changeTab, 'general')
+
+      await expect(page).toMatchElement('h1', { timeout: 5000 }) // Wait for tab to load
+      await expect(page).not.toMatchElement('h2[id^=_], h3[id^=_]', { timeout: 5000 }) // Beef
+
+      await page.evaluate(changeTab, 'geography')
+
+      await expect(page).toMatchElement('h1', { timeout: 5000 }) // Wait for tab to load
+      await expect(page).not.toMatchElement('h2[id^=_], h3[id^=_]', { timeout: 5000 }) // Beef
+
+      await page.evaluate(changeTab, 'keyboard')
+
+      await expect(page).toMatchElement('h1', { timeout: 5000 }) // Wait for tab to load
+      await expect(page).not.toMatchElement('h2[id^=_], h3[id^=_]', { timeout: 5000 }) // Beef
+
+      await page.evaluate(changeTab, 'math')
+
+      await expect(page).toMatchElement('h1', { timeout: 5000 }) // Wait for tab to load
+      await expect(page).not.toMatchElement('h2[id^=_], h3[id^=_]', { timeout: 5000 }) // Beef
+
+      await page.evaluate(changeTab, 'muzak')
+
+      await expect(page).toMatchElement('h1', { timeout: 5000 }) // Wait for tab to load
+      await expect(page).not.toMatchElement('h2[id^=_], h3[id^=_]', { timeout: 5000 }) // Beef
+
+      await page.evaluate(changeTab, 'physics')
+
+      await expect(page).toMatchElement('h1', { timeout: 5000 }) // Wait for tab to load
+      await expect(page).not.toMatchElement('h2[id^=_], h3[id^=_]', { timeout: 5000 }) // Beef
+
+      await page.evaluate(changeTab, 'programming')
+
+      await expect(page).toMatchElement('h1', { timeout: 5000 }) // Wait for tab to load
+      await expect(page).not.toMatchElement('h2[id^=_], h3[id^=_]', { timeout: 5000 }) // Beef
+    })
+  })
+
+  describe('Finnish and Swedish tabs have equal list of ids', () => {
+    const getCurrentIdList = (): string[] => {
+      const ids: string[] = []
+      document.querySelectorAll('h2[id], h3[id]').forEach(function (e) {
+        ids.push(e.getAttribute('id'))
+      })
+      return ids
+    }
+
+    const ifDuplicateEntry = (list: string[]): string[] => {
```

Comment:
Mitä tämän funktion on tarkoitus palauttaa?

Reply:
Se palauttaa listan sivun ID:stä, jos sivulla on useampia samoja ID:itä. Eli yritetään etsiä onko saadussa listassa samoja alkioita ja jos on, palautetaan lista. Listasta on helppo katsoa millä sivulla virhe on ja sortatusta listasta näkee mitä id on tuplana.